proposed/vcpkg.md

```diff
@@ -0,0 +1,118 @@
+- NuGet packages are not easy to consume from native CMake projects, so we end up having to wire up MSBuild to consume them and construct paths to pass to CMake.
+
+Additionally, [the VMR design documents on external source](https://github.com/dotnet/arcade/blob/main/Documentation/UnifiedBuild/VMR-Strategy-For-External-Source.md) defines that we shouldn't use submodules directly in our build.
+
+For native dependencies, we could alternatively use Vcpkg, a cross-platform C++ package manager provided by another team at Microsoft.
+Vcpkg would make it easier for us to both consume native dependencies in our projects, as many C++ libraries that we may consider using are already available in vcpkg.
+It would also make it easier for us to contribute our own native projects, like the nethost library, back to vcpkg in an official manner.
+
+## Requirements
+
+- We must be able to version and patch dependencies ourselves.
+- We must be able to produce private patches that we can consume without public publishing.
+- We must be able to consume dependencies from vcpkg in our projects.
+- It must be possible to use vcpkg in offline build scenarios.
+- It must be possible to use vcpkg in the VMR builds.
+
+### Goals
+
+- Enable developers of .NET to consume native dependencies from vcpkg.
+
+### Non-Goals
+
+- Replace all existing native dependency mechanisms.
+- Initially, using vcpkg from MSBuild is considered out of scope.
+- Using vcpkg in .NET 9 is out of scope.
+
+## Stakeholders and Reviewers
+
+- @jkoritzinsky
+- @mmitche
+- @jkotas
+- @AaronRobinsonMSFT
+- @wtgodbe
+
+## Design
+
+### Acquiring vcpkg
+
+Arcade should provide functionality in the scripts in `eng/common` to acquire the vcpkg tool.
+vcpkg will be able requested by an entry in the global.json file for a repository.
+The Arcade build scripts will define which version of vcpkg is used.
+
+As we will be using our own custom registry, Arcade can do a sparse checkout of the vcpkg repository to reduce the amount of data that needs to be downloaded, excluding the default registry.
+
+To support offline builds, the vcpkg repo and a matching version of the [vcpkg-tool repository](https://github.com/microsoft/vcpkg-tool) will be included as a submodule in the source-build-externals repository.
+
+### Custom registry
+
+We can either create a new repository for the registry or use the source-build-externals repository as a vcpkg registry as well. This registry will, similar to the `dotnet-public` NuGet feed, contain all of the public packages that we want to consume from within our projects. However, unlike the `dotnet-public` NuGet feed, we will need to modify the portfiles for the packages that we want to consume to not use online sources in an offline build. This likely means that for each package, we'll need to create a submodule in source-build-externals or another repository for each package that we want to consume.
```

Comment:
Curious about the following:

How often you see the versions of these packages changing?
You mentioned the importance of patching earlier.  Where would that be done if the source-build-externals(SBE) repo were used?  Would the patching be done in a dotnet fork of the original source and then SBE would reference the fork?

Reply:
We usually pull a new version of our dependencies about once a year, mainly to get the upstream version that has the patches that we made locally and sent to the upstream in the prior release.

For patches, we'd either patch in a dotnet fork (which SBE would then reference) or we'd use patch files which vcpkg can apply as part of the package configuration.